src/main/java/org/jenkinsci/plugins/workflow/actions/ArgumentsAction.java

```diff
@@ -253,4 +262,47 @@ public boolean isUnmodifiedArguments() {
         // Cacheable, but arguments lists will be quite short and this is unlikely to get invoked heavily.
         return checkArgumentsLackPlaceholders(this.getArgumentsInternal());
     }
+
+    /**
+     * Like {@link #getArguments(FlowNode)} but attempting to resolve actual classes.
+     * If you need to reconstruct actual classes of nested objects (for example to pass to {@link PluginManager#whichPlugin}),
+     * it is not trivial to get this information from the form in which they were supplied to {@link DescribableModel#instantiate}.
+     * For example, nested objects (where present and not a {@link NotStoredReason}) might have been
+     * <ul>
+     * <li>an {@link UninstantiatedDescribable}, if given by a symbol
+     * <li>a {@link Map}, if given by {@link DescribableModel#CLAZZ}
+     * <li>a {@link Describable}, if constructed directly (rare)
+     * </ul>
+     * This method will instead attempt to return a normalized tree using {@link UninstantiatedDescribable} in all those cases.
+     * You could use {@link UninstantiatedDescribable#getModel} (where available) and {@link DescribableModel#getType} to access live classes.
+     * Where information is missing, this will just return the best it can.
+     */
+    @Nonnull
+    public static Map<String, ?> getResolvedArguments(@Nonnull FlowNode n) {
+        ArgumentsAction aa = n.getPersistentAction(ArgumentsAction.class);
+        if (aa == null) {
+            return Collections.emptyMap();
+        }
+        Map<String, Object> args = aa.getArgumentsInternal();
+        if (n instanceof StepNode) {
+            StepDescriptor d = ((StepNode) n).getDescriptor();
+            if (d != null) {
+                try {
+                    return resolve(new DescribableModel<>(d.clazz), args).getArguments();
+                } catch (Exception x) { // all sorts of things could go wrong here
+                    LOGGER.log(Level.FINE, "coult not resolve " + args + " for " + d.clazz.getName(), x);
+                    // TODO this does not handle NotStoredReason’s well
+                    // more robust to recursively traverse the tree and use e.g. DescribableModel.resolveClass to populate details
```

Comment:
Made a brief attempt to do this but seemed too complicated to take on now, and would require some nontrivial API changes to `structs`; it is a corner case anyway.